### ISSUE TITLE
[java] EmptyMethodInAbstractClassShouldBeAbstract should consider Lombok's @Synchronized

### ISSUE THREAD

**Issue by Belle-PL** [pmd/pmd#4211]
**Affects PMD Version:6.51.0**

**Rule: EmptyMethodInAbstractClassShouldBeAbstract**

Please provide the rule name and a link to the rule documentation:
<https://pmd.github.io/latest/pmd_rules_java_codestyle.html#emptymethodinabstractclassshouldbeabstract>

**Description:**
In the following code example, PMD reports a warning in line 4, however, the annotation `@Synchronized` can implement a synchronized statement in the method, and `isVisible` is not empty actually. Hence, this is a false positive.

**Code Sample demonstrating the issue:**

```java
import lombok.Synchronized;
public abstract class Test {
    @Synchronized
    public void isVisible() {  // should not report a warning in this line
        ;
    }
}
```

**Expected outcome:**

PMD reports a violation at line 4, but that's wrong. That's a false positive.

**Running PMD through:** *CLI*

**References:**
* https://projectlombok.org/features/Synchronized

**Comment by adangel** [comment]
Thanks for the report. However, the method is still empty, so nothing is synchronized and the method can actually be removed.

Is the problem maybe, that you can't annotate an abstract method with Lombok's `@Synchronized`? Actually, this also make sense, since synchronization is typically considered an implementation detail - without knowing what `isVisible()` will do, it might or might not need synchronization. Therefore the annotation should be used on the concrete subclass.

I think, this is not a false positive. IMHO the method `isVisible` should be abstract and not annotated at all.
